csrc/executor.h

```diff
@@ -247,13 +275,49 @@ class FusionExecutor : public NonCopyable {
         compiled_kernel_->cubin, "-fun 1 -c");
   }
 
-  std::string getCanonicalKernelName() const {
-    return kernelNamespace() + "::" + kernelName();
+  static void setGlobalFusionCount(int64_t new_fusion_count) {
+    global_fusion_count_.store(new_fusion_count);
+  }
+
+  static int64_t getGlobalFusionCount() {
+    return global_fusion_count_.load();
+  }
+
+  void createKernelId(
+      ScheduleHeuristic heuristic = ScheduleHeuristic::None,
+      int64_t fusion_id = 0,
+      int64_t concrete_id = 0,
+      int64_t runtime_id = 0,
+      int64_t group_id = 0) {
+    NVF_ERROR(fusion_id > -1, "Invalid fusion_id.");
+    NVF_ERROR(concrete_id > -1, "Invalid concrete_id.");
+    NVF_ERROR(runtime_id > -1, "Invalid runtime_id.");
+    NVF_ERROR(group_id > -1, "Invalid group_id");
+
+    heuristic_ = heuristic;
+    fusion_id_ = fusion_id;
+    concrete_id_ = concrete_id;
+    runtime_id_ = runtime_id;
+    group_id_ = group_id;
+    ++global_fusion_count_;
+
+    std::stringstream ss;
+    if (isOptionEnabled(EnableOption::StaticFusionCount)) {
+      ss << global_fusion_count_.load();
+    } else {
+      ss << toString(heuristic_);
+      ss << "_f" << fusion_id_;
+      ss << "_c" << concrete_id_;
+      ss << "_r" << runtime_id_;
+      ss << "_g" << group_id_;
+    }
+    kernel_id_ = ss.str();
```

Comment:
Let'a add an API to check if kernel id has been created properly in this class.

Reply:
I added `bool validKernelId() const`

Reply:
any chance you forget to push the code? or github is just slow today...

Reply:
I got distracted and forgot to push the changes.